src/dotnet-svcutil/lib/src/Metadata/ServiceDescriptor.cs

```diff
@@ -116,27 +116,35 @@ public async Task ImportMetadataAsync(Action<WsdlImporter> onWsdlImporterCreated
                 Type type = assembly.GetType("Microsoft.Tools.ServiceModel.Svcutil.NamedPipeMetadataImporter");
                 if (type != null)
                 {
-                    object typeInstance = Activator.CreateInstance(type, null);
-                    MethodInfo methodInfo = type.GetMethod("GetMetadatadataAsync", BindingFlags.Public | BindingFlags.Instance);
-                    var xmlReader = await (Task<System.Xml.XmlReader>)methodInfo.Invoke(typeInstance, new object[] { MetadataUrl });
+                    try
+                    {
+                        object typeInstance = Activator.CreateInstance(type, null);
+                        MethodInfo methodInfo = type.GetMethod("GetMetadatadataAsync", BindingFlags.Public | BindingFlags.Instance);
```

Comment:
Is this necessary?  If necessary, at least add comment.

Reply:
It is necessary. As per our original plan, we are utilizing a reflection approach to download metadata for the named pipe service. The method entry point for this is GetMetadataAsync, defined within Microsoft.Svcutil.NamedPipeMetadataImporter.csproj.